spec/spec.md

```diff
@@ -1390,28 +1390,60 @@ format-related rules above:
     },
     "field": {
       "type": "object",
-      "properties": {
-        "path": {
-          "type": "array",
-          "items": { "type": "string" }
+      "oneOf": [
+        {
+          "properties": {
+            "path": {
+              "type": "array",
+              "items": { "type": "string" }
+            },
+            "purpose": { "type": "string" },
+            "filter": {
+              "type": "object",
+              "properties": {
+                "type": { "type": "string" },
+                "format": { "type": "string" },
+                "pattern": { "type": "string" },
+                "minimum": { "type": "string" },
+                "minLength": { "type": "integer" },
+                "maxLength": { "type": "integer" }
+              },
+              "required": ["type"],
+              "additionalProperties": false
+            }
+          },
+          "required": ["path"],
+          "additionalProperties": false
         },
-        "purpose": { "type": "string" },
-        "filter": {
-          "type": "object",
+        {
           "properties": {
-            "type": { "type": "string" },
-            "format": { "type": "string" },
-            "pattern": { "type": "string" },
-            "minimum": { "type": "string" },
-            "minLength": { "type": "integer" },
-            "maxLength": { "type": "integer" }
+            "path": {
+              "type": "array",
+              "items": { "type": "string" }
+            },
+            "purpose": { "type": "string" },
+            "filter": {
+              "type": "object",
+              "properties": {
+                "type": { "type": "string" },
+                "format": { "type": "string" },
+                "pattern": { "type": "string" },
+                "minimum": { "type": "string" },
```

Comment:
shouldn't this be a number?
```suggestion
                "minimum": { "type": "number" },
```

Reply:
right again

Reply:
it can actually be a string or integer, for example we could be comparing dates

Reply:
That makes sense. I've updated the suggestion above to reflect that.